src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/listingrequest/CreateListingRequestAction.java

```diff
@@ -84,8 +87,12 @@ public FactMessage generateFactMessage() {
     public List<UserTransactionData> generateUserTransactionData() {
         List<UserTransactionData> userTransactionDataList = new ArrayList<>();
 
-        userTransactionDataList.add(prepareCreateUserTransactionData("hearingPart", hearingPart.getId(), 0));
-        userTransactionDataList.add(prepareCreateUserTransactionData("hearing", hearing.getId(), 1));
+        int orderNr;
+        for (orderNr = 0; orderNr < hearingParts.size(); orderNr++) {
```

Comment:
Why should we do that? it doesn't really mater if one of the hearing parts were created before or after another one. There could be put with the same order but probably before hearing. Did you try to revert changes on creation?